keras/utils/vis_utils.py

```diff
@@ -13,7 +13,10 @@
 
 
 def _check_pydot():
-    if not (pydot and pydot.find_graphviz()):
+    try:
+        # Attempt to create an image of a blank graph to check the pydot/graphviz installation.
+        pydot.Dot.create(pydot.Dot())
+    except:
```

Comment:
For what it's worth I will probably also try submit a pull request to pyplot to raise a more appropriate exception.